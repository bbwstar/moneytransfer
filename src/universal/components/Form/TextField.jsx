import React from 'react';
import PropTypes from 'prop-types';

const TextField = props =>
  (<div className="field">
    <label className="label" htmlFor={props.name.toLowerCase()}>
      {props.name}
    </label>
    <p className="control">
      <input
        type="text"
        id={props.name.toLowerCase()}
        name={props.name.toLowerCase()}
        value={props.value}
        onChange={props.handleChange}
        className="input is-medium is-fullwidth"
      />
    </p>
  </div>);

TextField.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};
export default TextField;
