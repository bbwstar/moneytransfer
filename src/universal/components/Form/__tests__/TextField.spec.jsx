import React from 'react';
import { FormattedMessage } from 'react-intl';
import { mountWithIntl } from 'utils/intl-enzyme-test-helper';
import toJson from 'enzyme-to-json';

import TextField from '../TextField';

const handleChange = jest.fn();
const wrapper = mountWithIntl(
  <TextField
    name="test"
    nameFormattedMessage={<FormattedMessage id="testMessage" defaultMessage="testMessage" />}
    value="defaultValue"
    handleChange={handleChange}
  />,
);

describe('TextField (Snapshot)', () => {
  it('renders without exploding', () => {
    const tree = toJson(wrapper);

    expect(tree).toMatchSnapshot();
  });

  it('calls handleChange function on input change', () => {
    wrapper.find('input').simulate('change');

    expect(handleChange).toBeCalled();
  });
});
