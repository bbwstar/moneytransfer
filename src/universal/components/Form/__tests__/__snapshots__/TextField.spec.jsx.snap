// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField (Snapshot) renders without exploding 1`] = `
<TextField
  handleChange={[Function]}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  name="test"
  nameFormattedMessage={
    <FormattedMessage
      defaultMessage="testMessage"
      id="testMessage"
      values={Object {}}
    />
  }
  value="defaultValue"
>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="test"
    >
      <FormattedMessage
        defaultMessage="testMessage"
        id="testMessage"
        values={Object {}}
      >
        <span>
          testMessage
        </span>
      </FormattedMessage>
    </label>
    <p
      className="control"
    >
      <input
        className="input is-medium is-fullwidth"
        id="test"
        name="test"
        onChange={[Function]}
        type="text"
        value="defaultValue"
      />
    </p>
  </div>
</TextField>
`;
